paddleseg/models/losses/semantic_encode_cross_entropy_loss.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+from paddle import nn
+import paddle.nn.functional as F
+
+from paddleseg.cvlibs import manager
+
+
+@manager.LOSSES.add_component
+class SECrossEntropyLoss(nn.Layer):
+    """
+    The Semantic Encoding Loss implementation based on PaddlePaddle.
+
+    Args:
+        num_classes (int): The unique number of target classes.
+        ignore_index (int): Specifies a target value that is ignored
+            and does not contribute to the input gradient. Default: 255.
+    """
+    def __init__(self, num_classes, ignore_index=255):
+        super(SECrossEntropyLoss, self).__init__()
+        self.num_classes = num_classes
```

Comment:
why do we need to assign num_class by the user?  it equals to logit.shape[1] which can be read by us.

Reply:
modified